tests/Eloquent/CallBuilderTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Tests\Eloquent;
+
+use BadMethodCallException;
+use Generator;
+use MongoDB\Laravel\Eloquent\Builder;
+use MongoDB\Laravel\Query\Builder as QueryBuilder;
+use MongoDB\Laravel\Tests\Models\User;
+use MongoDB\Laravel\Tests\TestCase;
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\Attributes\Test;
+use RuntimeException;
+
+use function assert;
+
+final class CallBuilderTest extends TestCase
+{
+    #[Test]
+    #[Dataprovider('provideFunctionNames')]
+    public function callingABuilderMethodDoesNotReturnTheBuilderInstance(string $method, string $className, $parameters = []): void
+    {
+        $builder = User::query()->newQuery();
+        assert($builder instanceof Builder);
+
+        self::assertNotInstanceOf(expected: $className, actual: $builder->{$method}(...$parameters));
+    }
+
+    public static function provideFunctionNames(): Generator
+    {
+        yield 'does not exist' => [
+            'doesntExist',
+            Builder::class,
+        ];
+
+        yield 'get bindings' => [
+            'getBindings',
+            Builder::class,
+        ];
+
+        yield 'get connection' => [
+            'getConnection',
+            Builder::class,
+        ];
+
+        yield 'get grammar' => [
+            'getGrammar',
+            Builder::class,
+        ];
+
+        yield 'insert get id' => [
+            'insertGetId',
+            Builder::class,
+            [['user' => 'foo']],
+        ];
+
+        yield 'to Mql' => [
+            'toMql',
+            Builder::class,
+        ];
+    }
+
+    #[Test]
+    #[DataProvider('provideUnsupportedMethods')]
+    public function callingUnsupportedMethodThrowsAnException(string $method, string $exceptionClass, string $exceptionMessage, $parameters = []): void
+    {
+        $builder = User::query()->newQuery();
+        assert($builder instanceof Builder);
+
+        $this->expectException($exceptionClass);
```

Comment:
verify that certain methods are not supported, checking both the exception type and its message